.github/actions/native-build/action.yml

```diff
@@ -0,0 +1,162 @@
+        ccache -V
+        ccache -p
+        ccache -s
+        ccache -z # Zero stats before build
+        
+    # Make sure .ccache directory exists with proper permissions
+    - name: Setup ccache directory
+      shell: bash
+      run: |
+        mkdir -p ~/.ccache
+        mkdir -p android/.ccache
+        if [ -f "android/.ccache/ccache.conf" ]; then
+          cp android/.ccache/ccache.conf ~/.ccache/
+          echo "Using project-specific ccache config"
+        else
+          echo "No project-specific ccache config found, using defaults"
+        fi
+        chmod -R 755 ~/.ccache
+        ls -la ~/.ccache
+
+    # Extract external native build tasks for cache key
+    - name: Extract external native build tasks
+      shell: bash
+      run: |
+        cd android && \
+        cat gradle-tasks.txt | grep -i externalnative > external-native-tasks.txt
+        cat external-native-tasks.txt
+    
+    # Generate a hash of the external-native-tasks.txt file
+    - name: Generate native build hash
+      id: native-build-hash
+      shell: bash
+      run: |
+        if [ -f android/external-native-tasks.txt ]; then
+          HASH=$(sha256sum android/external-native-tasks.txt | cut -d ' ' -f 1)
+          echo "hash=$HASH" >> $GITHUB_OUTPUT
+          echo "Native build hash: $HASH"
+        else
+          echo "hash=nohash" >> $GITHUB_OUTPUT
+          echo "external-native-tasks.txt not found, using default hash"
+        fi
+      
+    # Check if op-sqlite native build artifacts exist
+    - name: Check if op-sqlite native build artifacts exist
+      id: check-op-sqlite-artifacts
+      uses: actions/download-artifact@v4
+      continue-on-error: true
+      with:
+        name: op-sqlite-native-build-${{ inputs.arch }}-${{ steps.native-build-hash.outputs.hash }}
+        path: op-sqlite-native-build-${{ inputs.arch }}.tar.gz
+
+    # Build native components if needed
+    - name: Build Native Components
+      id: build-native
+      if: inputs.skip_native_build != 'true' && steps.check-op-sqlite-artifacts.outputs.cache-hit != 'true'
+      env:
+        CCACHE_DEBUG: 1
+        CCACHE_VERBOSE: 1
+        CCACHE_LOGFILE: "/tmp/ccache.log"
+      run: |
+        echo "Building native components as op-sqlite artifacts were not available"
+        echo "Setting up environment variables for ccache:"
+        env | grep -i ccache
+        cd android && \
+        ./gradlew externalNativeBuildDebug \
+                  externalNativeBuildRelease \
+        --parallel --max-workers=4 --build-cache --configure-on-demand \
+        -Pandroid.prefabPublishing=true \
+        -Pandroid.prefabPackageType=STATIC \
+        -PBUILD_ARCH="${{ inputs.arch }}" \
+        -PreactNativeArchitectures="${{ inputs.arch }}" \
+        -x preBuild -x preDebugBuild -x preReleaseBuild
+      shell: bash
+
+    # Archive op-sqlite native build artifacts
+    - name: Archive op-sqlite native build artifacts
+      if: inputs.skip_native_build != 'true'
```

Comment:
tmate here run the build task again the see that shit works.

delete the dirs untar them and try again and see if it works